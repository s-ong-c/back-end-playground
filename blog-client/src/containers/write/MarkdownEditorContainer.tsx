import * as React from 'react';
import MarkdownEditor from '../../components/write/MarkdownEditor';
import { connect } from 'react-redux';
import { RootState } from '../../modules';
import {
  changeMarkdown,
  changeTitle,
  setHtml,
  convertEditorMode,
} from '../../modules/write';

import remark from 'remark';
import htmlPlugin from 'remark-html';
import breaks from 'remark-breaks';
import TagInputContainer from './TagInputContainer';
import WriteFooter from '../../components/write/WriteFooter';
interface OwnProps {}
interface StateProps {
  title: string;
  markdown: string;
}
interface DispatchProps {
  changeMarkdown: typeof changeMarkdown;
  changeTitle: typeof changeTitle;
  setHtml: typeof setHtml;
  convertEditorMode: typeof convertEditorMode;
}
export type MarkdownEditorContainerProps = OwnProps &
  StateProps &
  DispatchProps;

const MarkdownEditorContainer: React.SFC<MarkdownEditorContainerProps> = ({
  changeMarkdown,
  changeTitle,
  setHtml,
  convertEditorMode,
  title,
  markdown,
}) => {
  const onConvert = (markdown: string) => {
    remark()
      .use(breaks)
      .use(htmlPlugin)
      .process(markdown, (err: any, file: any) => {
        const html = String(file);
        setHtml(html);
        convertEditorMode();
      });
  };
  return (
    <MarkdownEditor
      title={title}
      markdown={markdown}
      onChangeMarkdown={changeMarkdown}
      onChangeTitle={changeTitle}
      onConvert={onConvert}
      tagInput={<TagInputContainer />}
      footer={<WriteFooter onPublish={() => {}} onTempSave={() => {}} />}
    />
  );
};

export default connect<StateProps, DispatchProps, OwnProps, RootState>(
  state => ({
    title: state.write.title,
    markdown: state.write.markdown,
  }),
  {
    changeMarkdown,
    changeTitle,
    setHtml,
    convertEditorMode,
  },
)(MarkdownEditorContainer);
