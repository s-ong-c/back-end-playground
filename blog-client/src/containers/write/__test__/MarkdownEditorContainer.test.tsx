import * as React from 'react';
import { render } from 'react-testing-library';
import MarkdownEditorContainer, {
  MarkdownEditorContainerProps,
} from '../MarkdownEditorContainer';
import { Provider } from 'react-redux';
import rootReducer from '../../../modules';
import { createStore } from 'redux';

describe('MarkdownEditorContainer', () => {
  const setup = (props: Partial<MarkdownEditorContainerProps> = {}) => {
    const store = createStore(rootReducer);
    const utils = render(
      <Provider store={store}>
        <MarkdownEditorContainer {...props} />
      </Provider>,
    );
    return {
      ...utils,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
