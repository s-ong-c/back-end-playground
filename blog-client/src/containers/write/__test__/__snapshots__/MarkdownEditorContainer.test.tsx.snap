// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownEditorContainer matches snapshot 1`] = `
<div>
  <div
    class="sc-jTzLTM iphFoa"
    data-testid="codemirror"
  >
    <div
      class="wrapper"
    >
      <div
        class="sc-jzJRlG cocNVz"
      >
        <textarea
          class="sc-bdVaJa kpOVWP"
          placeholder="제목을 입력하세요"
          style="height: 0px;"
        />
        <div
          class="sc-fjdhpX bOKVbc"
        />
        <div
          class="sc-kAzzGY iFQAgw"
        >
          <div
            class="sc-chPdSV jUtVBX"
            contenteditable="true"
            placeholder="#태그입력"
            tabindex="2"
          />
          <div
            class="sc-kGXeez BQOEH"
          >
            <div
              class="inside"
            >
              Enter 를 사용하여 태그를 사용해보세요!
              <br />
              등록된 태그를 클릭하면 삭제됩니다.
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bwzfXH knqGhr"
        id="toolbar"
      >
        <div
          class="sc-htpNat jgwOvs"
        >
          <button
            class="ql-header sc-ifAKCX bHEhhn"
            value="1"
          >
            <div
              class="sc-bxivhb cWJsna"
            >
              H
              <span>
                1
              </span>
            </div>
          </button>
          <button
            class="ql-header sc-ifAKCX bHEhhn"
            value="2"
          >
            <div
              class="sc-bxivhb cWJsna"
            >
              H
              <span>
                2
              </span>
            </div>
          </button>
          <button
            class="ql-header sc-ifAKCX bHEhhn"
            value="3"
          >
            <div
              class="sc-bxivhb cWJsna"
            >
              H
              <span>
                3
              </span>
            </div>
          </button>
          <button
            class="ql-header sc-ifAKCX bHEhhn"
            value="4"
          >
            <div
              class="sc-bxivhb cWJsna"
            >
              H
              <span>
                4
              </span>
            </div>
          </button>
        </div>
        <div
          class="sc-EHOje cbmVMA"
        />
        <div
          class="sc-htpNat jgwOvs"
        >
          <button
            class="ql-bold sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
              />
            </svg>
          </button>
          <button
            class="ql-italic sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
              />
            </svg>
          </button>
          <button
            class="ql-strike sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 19h4v-3h-4v3zM5 4v3h5v3h4V7h5V4H5zM3 14h18v-2H3v2z"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-EHOje cbmVMA"
        />
        <div
          class="sc-htpNat jgwOvs"
        >
          <button
            class="ql-blockquote sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
              />
            </svg>
          </button>
          <button
            class="ql-link sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
              />
            </svg>
          </button>
          <button
            class="ql-image sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </button>
          <button
            class="ql-code-block sc-ifAKCX bHEhhn"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
              />
            </svg>
          </button>
        </div>
        <div
          class="sc-EHOje cbmVMA"
        />
        <div
          class="sc-htpNat jgwOvs"
        >
          <button
            class="sc-ifAKCX bHEhhn"
            data-testid="quillconvert"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="sc-jzJRlG cocNVz"
      >
        <textarea
          id="bla"
          style="display: none;"
        />
        <div
          class="CodeMirror cm-s-one-light CodeMirror-wrap CodeMirror-empty"
        >
          <div
            style="overflow: hidden; position: relative; width: 3px; height: 0px;"
          >
            <textarea
              autocapitalize="off"
              autocorrect="off"
              spellcheck="false"
              style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
              tabindex="0"
            />
          </div>
          <div
            class="CodeMirror-vscrollbar"
            cm-not-content="true"
            tabindex="-1"
          >
            <div
              style="min-width: 1px;"
            />
          </div>
          <div
            class="CodeMirror-hscrollbar"
            cm-not-content="true"
            tabindex="-1"
          >
            <div
              style="height: 100%; min-height: 1px;"
            />
          </div>
          <div
            class="CodeMirror-scrollbar-filler"
            cm-not-content="true"
          />
          <div
            class="CodeMirror-gutter-filler"
            cm-not-content="true"
          />
          <div
            class="CodeMirror-scroll"
            tabindex="-1"
          >
            <div
              class="CodeMirror-sizer"
              style="margin-left: 0px;"
            >
              <div
                style="position: relative;"
              >
                <div
                  class="CodeMirror-lines"
                  role="presentation"
                >
                  <div
                    role="presentation"
                    style="position: relative; outline: none;"
                  >
                    <pre
                      class="CodeMirror-placeholder CodeMirror-line-like"
                      style="height: 0px; overflow: visible; direction: ltr;"
                    >
                      당신의 이야기를 적어보세요...
                    </pre>
                    <div
                      class="CodeMirror-measure"
                    >
                      <pre
                        class="CodeMirror-line-like"
                      >
                        <span>
                          xxxxxxxxxx
                        </span>
                      </pre>
                    </div>
                    <div
                      class="CodeMirror-measure"
                    />
                    <div
                      style="position: relative; z-index: 1;"
                    />
                    <div
                      class="CodeMirror-cursors"
                    />
                    <div
                      class="CodeMirror-code"
                      role="presentation"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              style="position: absolute; height: 30px; width: 1px;"
            />
            <div
              class="CodeMirror-gutters"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-cSHVUG fnIIPU"
      style="width: 0px;"
    >
      <div
        class="sc-kpOJdX bSGAhH"
      >
        <button
          class="sc-dxgOiQ gIXZQZ sc-gZMcBi gwJcWz"
          color="lightGray"
        >
          임시저장
        </button>
        <button
          class="sc-dxgOiQ gIXZQZ sc-gZMcBi fDcYxe"
          color="teal"
        >
          출간하기
        </button>
      </div>
    </div>
  </div>
</div>
`;
